src/js/utils/stream-time.js

```diff
@@ -0,0 +1,22 @@
+define([
+
+], function () {
+    function rewindPosition(amount, currentPosition, seekableRange, streamType) {
+        var pos = 0;
+        var rewindTarget = currentPosition - amount;
+
+        if (streamType === 'DVR') {
+            // When in DVR mode, give the seek position as the distance from the seekable range
+            // The provider will use this to calculate the real time to seek to
+            pos = Math.min(seekableRange - rewindTarget, seekableRange);
```

Comment:
What happens if `rewindTarget` is larger than `seekableRange`? .ie Someone makes an API call to seek 30s back but you only have 20s of buffer?